crates/contract/src/call.rs

```diff
@@ -271,8 +271,13 @@ impl<P: TempProvider, D: CallDecoder> CallBuilder<P, D> {
     }
 
     /// Uses a Legacy transaction instead of an EIP-1559 one to execute the call
-    pub fn legacy(self) -> Self {
-        todo!()
+    pub fn legacy(mut self) -> Self {
+        self.request.max_fee_per_blob_gas = None;
```

Comment:
unfortunately this does not add support for legacy transactions, `gas_price` still only sets EIP-1559 gas fields

Reply:
sorry - maybe I wasn't precise. the field `gas_price` is for legacy transactions, but the *method* `gas_price` only sets the fields `max_fee_per_gas` and `max_priority_fee_per_gas`. if the tx is a legacy tx *only* the `gas_price` field should be set, in all other cases `max_fee_per_gas` and `max_priority_fee_per_gas` should be set.

currently, if i do `tx.gas_price(..)` it sets the fields `max_fee_per_gas` and `max_priority_fee_per_gas`. the *field* `gas_price` is not set. if the tx is a legacy transaction, it should be. if it is not a legacy transaction, instead the `max_fee_*` fields should be set.

Reply:
> sorry - maybe I wasn't precise. the field `gas_price` is for legacy transactions, but the _method_ `gas_price` only sets the fields `max_fee_per_gas` and `max_priority_fee_per_gas`. if the tx is a legacy tx _only_ the `gas_price` field should be set, in all other cases `max_fee_per_gas` and `max_priority_fee_per_gas` should be set.
> 
> currently, if i do `tx.gas_price(..)` it sets the fields `max_fee_per_gas` and `max_priority_fee_per_gas`. the _field_ `gas_price` is not set. if the tx is a legacy transaction, it should be. if it is not a legacy transaction, instead the `max_fee_*` fields should be set.

thank you so much for this explain! Does this changes looks better to you ?